frontend/src/routes/candidate/(protected)/questions/[questionId]/+page.svelte

```diff
@@ -187,9 +187,24 @@
       return;
     }
 
+    // Check if all questions have been answered (before answer to current question is saved)
+    const allAnsweredBefore = $page.data.questions.every((question) =>
+      Object.keys(answerStore).includes(question.id.toString())
+    );
+
```

Comment:
I think this could be moved after the `await saveToServer();`, since the value isn't used until line 203.